src/unix/linux_like/linux/mod.rs

```diff
@@ -1206,6 +1206,81 @@ s! {
     }
 }
 
+cfg_if! {
```

Comment:
so yeah, this looks like the best I can do...

Defining a `PTP_CLOCK_CAPS_RSV_LEN` const just does not seem to work. not with cfg_if, a vanilly block with an if expression, defining a `const fn` and calling that: they all run into unreachable code in garando.

Next it tusrns out that `loongarch` is weird and special (or just has a later version maybe?), so that further complicates the situation. 

It looks like I can bring it down to three branches by just making the conditionals more complex (that almost adds as many lines as it saves...) Any other ideas?

Reply:
Where do all these definitions even come from? I don't see it in glibc or musl source so I assume it should be uapi source, but I only see the one definition with no conditions https://github.com/torvalds/linux/blob/43fb83c17ba2d63dfb798f0be7453ed55ca3f9c2/include/uapi/linux/ptp_clock.h#L94-L107

Reply:
_Oh_ all of this was just accounting for what we happen to test in CI rather than actual sources? Why didn't you just say so :) we should definitely only implement the latest and skip tests on platforms where we can explain the failure.

If this explains the platform-specific config for the other PTP structs (I think there was something for `sparc`), mind putting up a PR to drop that as well?

I suggested a few changes here but think this should mostly be good.

Reply:
the rest of the ptp structs are clean (no cfg's or anything like that). For the ioctl const functions there is some powerpc/sparc/mips-specific logic, but that really is because they define constants with different values. There are links to those values there so I think it's all good now.

Reply:
SGTM, I must have been thinking of something else. Thanks for the followup.